modules/dag-l0/src/main/scala/org/tessellation/dag/l0/domain/snapshot/programs/GlobalSnapshotEventCutter.scala

```diff
@@ -0,0 +1,118 @@
+package org.tessellation.dag.l0.domain.snapshot.programs
+
+import cats.MonadThrow
+import cats.effect.Async
+import cats.syntax.applicative._
+import cats.syntax.either._
+import cats.syntax.eq._
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.option._
+import cats.syntax.traverse._
+import cats.syntax.unorderedFoldable._
+
+import org.tessellation.dag.l0.infrastructure.snapshot.{DAGEvent, GlobalSnapshotEvent, StateChannelEvent}
+import org.tessellation.ext.crypto.RefinedHasher
+import org.tessellation.ext.kryo._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.node.shared.domain.event.EventCutter
+import org.tessellation.schema.Block
+import org.tessellation.security.hash._
+import org.tessellation.security.{Hashed, Hasher}
+
+import eu.timepit.refined.types.numeric.PosInt
+
+object GlobalSnapshotEventCutter {
+  def make[F[_]: Async: KryoSerializer: Hasher](maxBinarySizeBytes: PosInt): EventCutter[F, GlobalSnapshotEvent] =
+    new EventCutter[F, GlobalSnapshotEvent] {
+
+      def cut(events: Set[GlobalSnapshotEvent]): F[Set[GlobalSnapshotEvent]] =
+        needsCutting(events)
+          .ifM(
+            EventsState(events)
+              .map(Iterator.unfold(_)(_.next))
+              .flatMap(removeEvents(events, _)),
+            events.pure[F]
+          )
+
+      def needsCutting(events: Set[GlobalSnapshotEvent]): F[Boolean] =
+        events.toBinaryF.map(_.length > maxBinarySizeBytes.value)
+
+      def removeEvents(startingEvents: Set[GlobalSnapshotEvent], iterator: Iterator[EventWithFee]): F[Set[GlobalSnapshotEvent]] =
+        startingEvents.tailRecM { events =>
+          for {
+            eventToRemove <- MonadThrow[F].fromOption(iterator.nextOption(), new IllegalStateException("No GlobalSnapshotEvent events"))
+            reducedEvents = events - eventToRemove.toGlobalSnapshotEvent
+            cutAgain <- needsCutting(reducedEvents)
+          } yield Either.cond(!cutAgain, reducedEvents, reducedEvents)
+        }
+
+      trait EventWithFee {
+        def fee: Long
+        def hash: Hash
+        def toGlobalSnapshotEvent: GlobalSnapshotEvent
+      }
+
+      val orderingEventWithFee: Ordering[EventWithFee] = Ordering.by[EventWithFee, (Long, Hash)](e => (e.fee, e.hash))
+
+      case class StateChannelEventWithFee(event: StateChannelEvent, hash: Hash) extends EventWithFee {
+        val fee: Long = event.snapshotBinary.value.fee.value.value
+
+        def toGlobalSnapshotEvent: GlobalSnapshotEvent = event.asLeft[DAGEvent]
+
+        def isMyChild(that: StateChannelEventWithFee): Boolean =
+          that.event.snapshotBinary.lastSnapshotHash === hash
+      }
+      object StateChannelEventWithFee {
+        def apply(event: StateChannelEvent): F[StateChannelEventWithFee] =
+          event.snapshotBinary.content.hash.map(StateChannelEventWithFee(event, _))
```

Comment:
Look at `StateChannelSnapshotService` and `consume` method to see how the hash is calculated for snapshot binary. Right now the `isMyChild` wouldn't ever match any two snapshots.